### DISCUSSION TITLE
working with gitea/forgego/codeberg

### DISCUSSION THREAD

**Discussion by mooseyboots** [magit/forge#557] [General]
i have been trying to get the gitea support going from the idea branch, but i'm unsure if it is ready or not quite?

i am able to run most of the functions, like `forge--gtea-get`, but when i call `forge-pull`, the command runs successfully but nothing is added to the magit display buffer.

i'm on debian/emacs 28, with up-to-date magit/forge installed via straight.el.

don't want to hassle, but if it should be working then i'd like to try it out.

**Comment by tarsius** [comment]
Currently only github and gitlab are fully supported. See https://magit.vc/manual/forge/Supported-Forges-and-Hosts.html.

**Comment by mooseyboots** [comment]
I'm aware of that, yes. But elsewhere your docs also state that contributions adding support for other forges were welcome. I was wondering if that unmerged code would be something I or others might hack on, even just to get some basics.

**Reply by tarsius** [reply]
Yes, https://github.com/magit/forge/pull/431.

I mention there that I would be working on it myself and I did. While doing so I came to the conclusion that instead of duplicating existing but questionable code, I would rather improve the existing code, and then copy/use that.

But there is a lot of other work that keeps me from doing that. Implementing support for other forges is very low priority for me still.

I might have to revisit the docs that say that contributions in this area are welcome. You are of course welcome to implement support for another forge in an extension, but I do not currently plan to merge such additions.
